### PR TITLE
Add mocking of some s3 glacier capabilities

### PR DESCRIPTION
**Pull request by netsettler**
* In ``dcicutils.command_utils``:

  * New decorator ``require_confirmation``

* In ``dcicutils.common``:

  * New variable ``ALL_S3_STORAGE_CLASSES``
  * New variable ``AVAILABLE_S3_STORAGE_CLASSES``
  * New variable ``S3_GLACIER_CLASSES``
  * New type hint ``S3GlacierClass``
  * New type hint ``S3StorageClass``

* In ``dcicutils.misc_utils``:

  * New function ``INPUT``
  * New function ``future_datetime``
  * New decorator ``managed_property``
  * New function ``map_chunked``
  * New function ``format_in_radix``
  * New function ``parse_in_radix``

* In ``dcicutils.qa_checkers``:

  * Fix bug in ``print`` statement recognizer

* In ``dcicutils.qa_utils``:

  * Support for Glacier-related operations in ``MockBotoS3Client``:

    * Method ``copy_object``
    * Method ``delete_object``
    * Method ``list_object_versions``
    * Method ``restore_object``

* Load ``coveralls`` dependency only dynamically in GA workflow, not in poetry, because it implicates ``docopt`` library, which needs ``2to3``, and would fail.

### CODE REVIEW

**Review by willronchetti — APPROVED** [review]
One meaningful typo but other than that looks good. I'll make any further changes needed once in my branch.

File: dcicutils/common.py [hunk 1]
```diff
@@ -67,3 +67,52 @@
...
+# fragile. In practice, we just want to not overly warn about normal kinds of storage.
+
+ALL_S3_STORAGE_CLASSES = [
+    'STANDARD', 'REDUCED_REDUNDANCY', 'STANDARD_IA', 'ONEZONE_IA', 'INTELLIGENT_TIERING',
+    'GLACIER', 'DEEP_ARCHIVE', 'OUTPOSTS', 'GLACIER_IR',
+]
+
+AVAILABLE_S3_STORAGE_CLASSES = [
+    'STANDARD', 'STANDARD_IA', 'INTELLIGENT_TIERING'
+]
```
[the remark this answers is not in the document]
Comment (netsettler) [reply dcicutils/common.py:91]: This is a conservatively chosen set.  We could just use the glacier set and its complement, I guess.

File: dcicutils/qa_utils.py [hunk 2]
```diff
@@ -2040,14 +2299,26 @@ def head_object(self, Bucket, Key, **kwargs):  # noQA - AWS argument naming styl
...
+                'StorageClass': attribute_block.storage_class,  # self._object_storage_class(filename=pseudo_filename)
                 # Numerous others, but this is enough to make the dictionary non-empty and to satisfy some of our tools
             }
+            restoration = attribute_block.restoration
+            if restoration:
+                assert isinstance(restoration, MockTemporaryRestoration)
+                if restoration.is_active():
+                    result['Restore'] = f'ongoing-request="false", expiry-date="{restoration.available_until}"'
+                else:
+                    result['Restora'] = f'ongoing-request="true"'
```
[the remark this answers is not in the document]
Comment (netsettler) [reply dcicutils/qa_utils.py]: Typo corrected and uploaded.

File: dcicutils/common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (netsettler) [reply dcicutils/common.py:91]: By the way, I think I ported this list from a SubmitCGAP branch (that we may not have merged) where we were running into problems about these not being immediately available, so I needed the list. So if you make changes to this, you'll want to cross-check with me about that branch.

[the remark this answers is not in the document]
Comment (netsettler) [reply dcicutils/common.py:91]: If there are other classes that act as available and get added to this list, that's probably not going to perturb anything. If there is a thought that we should just list the glacier states and not use this list, that's what might be incompatible without changing something. Then again, the change isn't hard and would show up on testing of a new version.

**Pull request merged**